initializer.go

```diff
@@ -35,26 +36,38 @@ func (w *metricDefWrapper) List(ctx context.Context, resourceID string, options
 }
 
 // CreateAzureClients creates Azure clients with service principal credentials
-func CreateAzureClients(subscriptionID string, clientID string, clientSecret string, tenantID string) (*AzureClients, error) {
-	credential, err := azidentity.NewClientSecretCredential(tenantID, clientID, clientSecret, nil)
+func CreateAzureClients(subscriptionID string, clientID string, clientSecret string, tenantID string, clientOptions *azcore.ClientOptions) (*AzureClients, error) {
```

Comment:
this will break all the users without bumping semver. Argument should either be optional or  it should be a new `CreateAzureClientsWithOpts` function

Reply:
I have made the parameter optional.